registry/handlers/app.go

```diff
@@ -374,14 +374,35 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 	})
 }
 
+// identifyUser attempts to resolve a username from the context and request. If
+// a username cannot be resolved, one is created from the remote IP
+func identifyUser(ctx context.Context, r *http.Request) string {
+	username := getUserName(ctx, r)
+	if username == "" {
+		username = fmt.Sprintf("user@%s", ctxu.RemoteAddr(r))
+	}
+	return username
+}
+
+// WithUser adds the username to the context
+func WithUser(ctx context.Context, user interface{}) ctxu.Context {
+	username, ok := user.(string)
+	if !ok {
+		username = ""
+	}
+	return auth.WithUser(ctx, auth.UserInfo{Name: username})
+}
+
 // context constructs the context object for the application. This only be
 // called once per request.
 func (app *App) context(w http.ResponseWriter, r *http.Request) *Context {
 	ctx := ctxu.WithRequest(app, r)
 	ctx, w = ctxu.WithResponseWriter(ctx, w)
+	ctx = WithUser(ctx, identifyUser(ctx, r))
```

Comment:
Is this needed? Once the authorization context runs, you can push a new logger on the context that will see "auth.user.name". This can be done after `app.authorize` is called in `app.dispatch`.